yt_dlp/extractor/yandexvideo.py

```diff
@@ -186,7 +188,21 @@ def _real_extract(self, url):
         return self.url_result(data_json['video']['url'])
 
 
-class ZenYandexIE(InfoExtractor):
+class ZenYandexBaseIE(InfoExtractor):
+    def _fetch_ssr_data(self, url, video_id):
+        webpage = self._download_webpage(url, video_id)
+        redirect = self._search_json(r'(?:var|let|const)\s+it\s*=', webpage, 'redirect', id, default={}).get('retpath')
+        if redirect:
+            video_id = self._match_id(redirect)
```

Comment:
shouldn't we return the new `video_id` too?

Reply:
Ah, good catch. I oversaw that.